auditor/src/client/mod.rs

```diff
@@ -156,33 +217,53 @@ impl AuditorClient {
         }
     }
 
+    /// Update an existing record in the Auditor instance.
+    ///
+    /// # Errors
+    ///
+    /// * [`ClientError::ReqwestError`] - If there was an error sending the HTTP request.
     #[tracing::instrument(
         name = "Sending a record update to AUDITOR server.",
         skip(self, record),
         fields(record_id = %record.record_id)
     )]
-    pub async fn update(&self, record: &RecordUpdate) -> Result<(), reqwest::Error> {
+    pub async fn update(&self, record: &RecordUpdate) -> Result<(), ClientError> {
         self.client
             .post(&format!("{}/update", &self.address))
             .header("Content-Type", "application/json")
             .json(record)
             .send()
-            .await?
-            .error_for_status()?;
+            .await
+            .map_err(ClientError::ReqwestError)?
+            .error_for_status()
+            .map_err(ClientError::ReqwestError)?;
         Ok(())
     }
 
+    /// Gets all records from the Auditors database.
+    ///
+    /// # Errors
+    ///
+    /// * [`ClientError::ReqwestError`] - If there was an error sending the HTTP request.
     #[tracing::instrument(name = "Getting all records from AUDITOR server.", skip(self))]
-    pub async fn get(&self) -> Result<Vec<Record>, reqwest::Error> {
+    pub async fn get(&self) -> Result<Vec<Record>, ClientError> {
         self.client
             .get(&format!("{}/get", &self.address))
             .send()
-            .await?
-            .error_for_status()?
+            .await
+            .map_err(ClientError::ReqwestError)?
+            .error_for_status()
+            .map_err(ClientError::ReqwestError)?
             .json()
             .await
+            .map_err(ClientError::ReqwestError)
```

Comment:
Mapping the errors so often doesn't look great, but I'm not sure if there is a better solution. The only thing I can think of is to create a private function which contains the actual code an returns a `Result<Vec<Record>, request::Error>`. Then `get` would only call that function and map the error. Not sure if that is good style. Applies also to the following code.

Reply:
As discussed in person, we will stay with the current solution for now, because we are not really if it makes sense to continue with the private function approach.

Reply:
I've added trait implementations of `From<reqwest::Error>` and `From<OutOfRangeError>` for `ClientError`, which reduces the usage of `map_err()` calls. See f1c1ff9